test/e2e/specs/specs-playwright/wp-signup__screenshot.ts

```diff
@@ -0,0 +1,284 @@
+			await navbarCompnent.clickMySites();
+			const sidebarComponent = new SidebarComponent( page );
+			await sidebarComponent.navigate( 'Upgrades', 'Plans' );
+		} );
+
+		it( 'Click on the "Plans" navigation tab', async function () {
+			plansPage = new PlansPage( page );
+			await plansPage.clickTab( 'Plans' );
+		} );
+
+		it( 'Click on "Upgrade" button for WordPress.com Business plan', async function () {
+			await plansPage.clickPlanActionButton( { plan: 'Business', buttonText: 'Upgrade' } );
+		} );
+
+		it( 'WordPress.com Business is added to cart', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( cartItemForBusinessPlan );
+		} );
+
+		it( 'Screenshot checkout page for all en and non-en locales', async function () {
+			const changeUILanguageFlow = new ChangeUILanguageFlow( page );
+			for ( const locale of [ ...magnificientNonEnLocales, 'en' ] ) {
+				page.setViewportSize( { width: 1280, height: 720 } );
+				await page.goto( DataHelper.getCalypsoURL( 'home' ), { waitUntil: 'networkidle' } );
+				await changeUILanguageFlow.changeUILanguage( locale as LanguageSlug );
+				await page.goto( DataHelper.getCalypsoURL( 'home' ), { waitUntil: 'networkidle' } );
+				await cartCheckoutPage.visit( blogName );
+				const paymentDetails = DataHelper.getTestPaymentDetails();
+				await cartCheckoutPage.enterBillingDetails( paymentDetails );
+				await cartCheckoutPage.validatePaymentForm();
+				await page.screenshot( {
+					path: `tos_checkout_desktop_${ locale }.png`,
+					fullPage: true,
+					type: 'jpeg',
+					quality: 20,
+				} );
+				page.setViewportSize( { width: 410, height: 820 } );
+				await page.screenshot( {
+					path: `tos_checkout_mobile_${ locale }.png`,
+					fullPage: true,
+					type: 'jpeg',
+					quality: 20,
+				} );
+				page.setViewportSize( { width: 1024, height: 1366 } );
+				await page.screenshot( {
+					path: `tos_checkout_tablet_${ locale }.png`,
+					fullPage: true,
+					type: 'jpeg',
+					quality: 20,
+				} );
+			}
+		} );
+
+		it( 'Zip screenshots and upload', async function () {
+			const zipFilename = 'tos-screenshots.zip';
+			const archive = archiver( 'zip', {
+				zlib: { level: 9 }, // Sets the compression level.
+			} );
+			const output = fs.createWriteStream( zipFilename );
+			archive.pipe( output );
+			archive.glob( 'tos_*' );
+			archive.finalize();
+
+			output.on( 'close', function () {
+				const form = new FormData();
+				const bearerToken = DataHelper.getTosUploadToken();
+				form.append( 'zip_file', fs.createReadStream( zipFilename ) );
+				fetch( 'https://public-api.wordpress.com/wpcom/v2/screenshots', {
+					method: 'POST',
+					body: form,
+					headers: {
+						Authorization: `Bearer ${ bearerToken }`,
+					},
+				} )
+					.then( ( response ) => response.json() )
+					.then( ( response ) => expect( response?.upload_status ).toStrictEqual( 'success' ) );
```

Comment:
For example, if this task can be performed via cURL, then it can be moved out of this script and into a build step, which would:
1. allow for multiple scripts to generate artifacts in parallel, then zip into one archive at end.
2. better control over where the artifacts end up (on TeamCity or on the MarTech servers).